packages/data-stores/src/reader/contexts/site-subscriptions-query-props.tsx

```diff
@@ -19,10 +19,11 @@ export const SiteSubscriptionsQueryProps = createContext< SiteSubscriptionsQuery
 	setFilterOption: () => undefined,
 } );
 
-export const SiteSubscriptionsQueryPropsProvider: React.FC< { children: React.ReactNode } > = ( {
-	children,
-} ) => {
-	const [ searchTerm, setSearchTerm ] = useState( '' );
+export const SiteSubscriptionsQueryPropsProvider: React.FC< {
+	initialSearchTerm?: string | ( () => string );
```

Comment:
The `initialSearchTerm` signature was unexpected. I expected to see a string argument, but saw a callback. Can we either use a string or a callback, and reflect that in the name?

Reply:
The intent is to exactly mirror the `initialState` argument that the native `useState` hook supports. In that sense it should be perfectly intuitive and expected, because folks already know how `useState` works 🙂 I don't mind clarifying that in the name (should it be `initialSearchTermState`?) or in a comment. Do you have any suggestions?